blockchain/v2/scheduler.go

```diff
@@ -8,8 +8,88 @@ import (
 	"time"
 
 	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
 )
 
+// Events
+
+// XXX: The handle API would be much simpler if it return a single event, an
+// Event, which embeds a terminationEvent if it wants to terminate the routine.
+
+type scFinishedEv struct {
+	priorityNormal
+}
+
+type scPrunePeerEv struct {
+	priorityHigh
+	peerID p2p.ID
+	reason error
+}
+
+type scschedulerrFailure struct {
+	priorityHigh
+	peerID p2p.ID
+	time   time.Time
+	reason error
+}
+
+// XXX: unify with bcBlockResponse
+type tmpBlockResponse struct {
```

Comment:
```suggestion
type bcBlockResponse struct {
```
processor.go should use scBlockReceived maybe?